src/glyphs.cpp

```diff
@@ -5,203 +5,620 @@
+    }
 
-    // node::ObjectWrap uses the first internal field to store the wrapped pointer.
-    tpl->InstanceTemplate()->SetInternalFieldCount(1);
-    tpl->SetClassName(name);
+    if (args.Length() < 4 || !args[3]->IsFunction()) {
+        return NanThrowTypeError("Callback must be a function");
+    }
 
-    // Add all prototype methods, getters and setters here.
-    NODE_SET_PROTOTYPE_METHOD(tpl, "serialize", Serialize);
-    NODE_SET_PROTOTYPE_METHOD(tpl, "range", Range);
-    NODE_SET_PROTOTYPE_METHOD(tpl, "codepoints", Codepoints);
+    v8::Local<v8::Function> callback = args[3].As<v8::Function>();
 
-    // This has to be last, otherwise the properties won't show up on the
-    // object in JavaScript.
-    target->Set(name, constructor->GetFunction());
+    RangeBaton* baton = new RangeBaton();
+    baton->file_name = *NanUtf8String(args[0]);
+
+    baton->start = args[1]->IntegerValue();
+    baton->end = args[2]->IntegerValue();
+
+    unsigned array_size = baton->end - baton->start;
+    for (unsigned i=baton->start; i <= array_size; i++) {
+        baton->chars.emplace_back(i);
+    }
+
+    baton->request.data = baton;
+    NanAssignPersistent(baton->callback, callback.As<v8::Function>());
+
+    uv_queue_work(uv_default_loop(), &baton->request, RangeAsync, (uv_after_work_cb)AfterRange);
+    NanReturnUndefined();
 }
 
-NAN_METHOD(Glyphs::New) {
-    NanScope();
+void LoadAsync(uv_work_t* req) {
+    LoadBaton* baton = static_cast<LoadBaton*>(req->data);
 
-    if (!args.IsConstructCall()) {
-        return NanThrowTypeError("Constructor must be called with new keyword");
+    FT_Library library = nullptr;
+    FT_Error error = FT_Init_FreeType(&library);
+    if (error) {
+        /* LCOV_EXCL_START */
+        baton->error_name = std::string("could not open FreeType library");
+        return;
+        /* LCOV_EXCL_END */
     }
-    if (args.Length() > 0 && !node::Buffer::HasInstance(args[0])) {
-        return NanThrowTypeError("First argument may only be a buffer");
+    FT_Face ft_face = 0;
+    int num_faces = 0;
+    for ( int i = 0; ft_face == 0 || i < num_faces; ++i )
+    {
+        FT_Error face_error = FT_New_Face(library, baton->file_name.c_str(), i, &ft_face);
+        if (face_error) {
+            baton->error_name = std::string("could not open Face") + baton->file_name;
+            return;
+        }
+        std::vector<int> points;
+        if (num_faces == 0)
+            num_faces = ft_face->num_faces;
+        FT_ULong charcode;
+        FT_UInt gindex;
+        charcode = FT_Get_First_Char(ft_face, &gindex);
+        while (gindex != 0) {
+            charcode = FT_Get_Next_Char(ft_face, charcode, &gindex);
+            if (charcode != 0) points.emplace_back(charcode);
+        }
+
+        std::sort(points.begin(), points.end());
+        auto last = std::unique(points.begin(), points.end());
+        points.erase(last, points.end());
```

Comment:
Should be pretty quick to use a `std:set` here instead https://github.com/mapbox/node-fontnik/issues/67